packages/shared/src/components/streak/ReadingStreakButton.tsx

```diff
@@ -47,11 +49,20 @@ function CustomStreaksTooltip({
 export function ReadingStreakButton({
   streak,
 }: ReadingStreakButtonProps): ReactElement {
+  const { trackEvent } = useAnalyticsContext();
   const isLaptop = useViewSize(ViewSize.Laptop);
   const [shouldShowStreaks, setShouldShowStreaks] = useState(false);
   const hasReadToday =
     new Date(streak.lastViewAt).getDate() === new Date().getDate();
 
+  useEffect(() => {
+    if (shouldShowStreaks) {
+      trackEvent({
+        event_name: AnalyticsEvent.OpenStreaks,
+      });
+    }
+  }, [shouldShowStreaks, trackEvent]);
```

Comment:
We usually do some checks on these cases when tracking events to make sure this function doesn't cause any re-render and resend of track event. Just making sure we're fine like this. 
This is what I would usually expect, but commenting since I've seen it being checked more.

Reply:
Yeah think in this case on click makes most sense actually 🙏

Reply:
Fair point, thought about using a click handler initially but since `trackEvent` is attached to a context, use effect would not trigger re-renders based on it here. I'll change it to your suggestion 👍